components/forms/select-input/select-input.js

```diff
@@ -0,0 +1,134 @@
+function SelectInput(id, options, validationMsg, placeholder, isRequired) {
+  let selectInput
+  let selectElement
+  let validationMsgElement
+  let isOpen = false
+  let displayValidationMsg = false
+  const _options = options
+  const _validationMsg = validationMsg
+  const _placeholder = isRequired ? placeholder + '*' : placeholder
+  let _valid = false
+  let _value = 'not set'
+  let _index = 0
+  let shouldPopulateList = true
+
+  const cacheDom = function() {
+    selectInput = document.getElementById(id)
+    selectElement = selectInput.querySelector('.msds-select-input__select')
+    validationMsgElement = selectInput.querySelector('.msds-select-input__validation-msg')
+  }
+
+  const bindEvents = function() {
+    selectElement.addEventListener('change', () => render())
+  }
+
+  const render = function() {
+    _index = selectElement.value
+    validationMsgElement.innerHTML = _validationMsg
+    if (isOpen) {
+      selectInput.classList.toggle('msds-select-input--open')
+    }
+    if (_validationMsg != '' && displayValidationMsg == true) {
+      validationMsgElement.style.display = 'block'
+    } else {
+      validationMsgElement.style.display = 'none'
+    }
+    if (_index == 0) {
+      selectElement.classList.add('msds-select-input__select--no-selection')
+    } else {
+      selectElement.classList.remove('msds-select-input__select--no-selection')
+      displayValidationMsg = false
```

Comment:
displayValidationMsg is set to false here but displayValidationMsg == true is used in the condition few lines before. It means that validation message disappear after next execution of render method and even more, if displayValidationMsg will be set to true before next execution of render() then displayValidationMsg never disappear even you set  displayValidationMsg = false.

If you want to hide displayValidationMsg in the same execution of render method you should move entire

if (_index == 0) {
....

condition just after line with setting index:
 _index = selectElement.value

Reply:
Also this condition is a good place to define displaying of validation message (now validation message is hide only) instead of exposing this logic for parent in validateComponent method. If you want to have it in separated method you can do it and then please invoke this method here. Also, validation logic should include isRequired boolean.

Reply:
im not quite sure what you mean here pawel. can we have a chat about it ?

Reply:
Sure, just let me know when you have a time :)